static/js/composebox_typeahead.js

```diff
@@ -212,6 +217,15 @@ exports.tokenize_compose_str = function (s) {
     while (i > min_i) {
         i -= 1;
         switch (s[i]) {
+            case '`':
+            case '~':
+                // Code block must start on a new line
+                if (i === 2) {
+                    return s.slice(0);
+                } else if (s[i-3] === "\n") {
+                    return s.slice(i-2);
```

Comment:
True, `i-3` would be negative if you'd only typed one or two backticks. Maybe charAt(i-3) would be safer? (Because charAt is defined to just return an empty string if you specify an index that doesn't exist)